src/indexed.js

```diff
@@ -0,0 +1,32 @@
+import {Seq} from 'immutable';
+import Base from './base';
+import {updateCursor} from './utils';
+
+function Indexed(root, data, keyPath, onChange, size) {
```

Comment:
Do I hear multiple inheritance? :) Would it be a better idea to compose and delegate? 

I don't know exactly what `class` does behind the scenes, but I think it's still prototypal inheritance, just limited to single inheritance. Might be wrong about that.